Take columns into account when resetting cache

NWuensche opened the pull request:
# Acknowledgments
Please check the following boxes with an `x` if they apply:
* [x] I have read the recent version of the [contribution guide](
  https://github.com/EventFahrplan/EventFahrplan/blob/master/CONTRIBUTING.md) before creating this pull request.
* [x] I have checked a few other pull requests to see how they are written.
* [x] The feature I want to propose would be useful for the majority of users, not only for me personally.
* [x] I am aware that EventFahrplan is mostly developed by one person in their unpaid spare time.
* [x] I can help myself to get this feature implemented or know someone who wants to do it.

# Description
- Also reset the hash-map when columns change to prevent showing old data
- Take columns into account when resetting cache.
- Tested on a Pixel 9a with Android 16

# Before
<!--
- Describe the behavior before the change.
- Add screenshot(s).
-->

Issue #826 is there

# After
<!--
- Describe the behavior after the change.
- Add screenshot(s).
-->

Assure #826 is done :D

---
<!-- Remove it this pull request does not resolve an issue -->
Resolves #826

NWuensche commented:
> Adding and removing an Engelsystem URL must work as expected regardless how often users put or remove it. ❌

Are you sure this is not a separate issue? I have this issue also in commit 480a6ddb9f8c3916ad30277338ab9a0fdd6227e0 ,
which is one before your mentioned "bug commit" https://github.com/EventFahrplan/EventFahrplan/pull/790 / 2f4a47df4da3d580738a0207ba3b4dd42343e8ff

johnjohndoe commented:
I am currently working on https://github.com/EventFahrplan/EventFahrplan/issues/784. Maybe wait until I fixed the bug, then the behavior should be clearer.

NWuensche commented:
> Adding and removing an Engelsystem URL must work as expected regardless how often users put or remove it. ❌

I fixed this issue now. However, my solution is completely unrelated to any of the cache-code I've modified before. I hope it's still ok.  

> Document at which devices/emulators the behavior has been tested. ❌

I've also written it into the description of this ticket now, I've tested it on a Pixel 9a.
